test/ng/compileSpec.js

```diff
@@ -1201,41 +1201,50 @@ describe('$compile', function() {
           });
         });
 
-        it('should fail if replacing and template doesn\'t have a single root element', function() {
-          module(function() {
-            directive('noRootElem', function() {
-              return {
-                replace: true,
-                template: 'dada'
-              };
-            });
-            directive('multiRootElem', function() {
-              return {
-                replace: true,
-                template: '<div></div><div></div>'
-              };
-            });
-            directive('singleRootWithWhiteSpace', function() {
+        describe('replace and not exactly one root element', function() {
+          var templateVar;
+
+          beforeEach(module(function() {
+            directive('template', function() {
               return {
                 replace: true,
-                template: '  <div></div> \n'
+                template: function() {
+                  return templateVar;
+                }
               };
             });
-          });
+          }));
 
-          inject(function($compile) {
-            expect(function() {
-              $compile('<p no-root-elem></p>');
-            }).toThrowMinErr('$compile', 'tplrt', 'Template for directive \'noRootElem\' must have exactly one root element. ');
+          they('should throw if: $prop',
+            {
+              'no root element': 'dada',
+              'multiple root elements': '<div></div><div></div>'
+            }, function($prop) {
 
-            expect(function() {
-              $compile('<p multi-root-elem></p>');
-            }).toThrowMinErr('$compile', 'tplrt', 'Template for directive \'multiRootElem\' must have exactly one root element. ');
+              inject(function($compile, $templateCache, $rootScope) {
+                templateVar = $prop;
+                expect(function() {
+                  $compile('<p template></p>');
+                }).toThrowMinErr('$compile', 'tplrt', 'Template for directive \'template\' must have exactly one root element.');
+              });
+          });
 
-            // ws is ok
-            expect(function() {
-              $compile('<p single-root-with-white-space></p>');
-            }).not.toThrow();
+          they('should not throw if the root element is accompanied by: $prop',
+            {
+              'whitespace': '  <div>Hello World!</div> \n',
+              'comments': '<!-- oh hi --><div>Hello World!</div> \n',
+              'comments + whitespace': '  <!-- oh hi -->  <div>Hello World!</div>  <!-- oh hi -->\n'
+            }, function($prop) {
+
+              inject(function($compile, $templateCache, $rootScope) {
+                templateVar = $prop;
+                var element;
+                expect(function() {
+                  element = $compile('<p template></p>')($rootScope);
+                }).not.toThrowMinErr('$compile', 'tplrt', 'Template for directive \'template\' must have exactly one root element.');
```

Comment:
Ok, but I think we should match against the error key (tplrt), because that is unlikely to change and we want to make sure that exact error is thrown.